src/tests/Parser/Brewmaster/Energy.test.js

```diff
@@ -0,0 +1,112 @@
+import Energy from 'Parser/BrewmasterMonk/Modules/Features/Energy';
+import { processEvents } from './Fixtures/processEvents';
+import { EnergyFight, buffs } from './Fixtures/EnergyFight';
+
+describe('Brewmaster.Energy', () => {
+  let energy;
+  beforeEach(() => {
+    energy = new Energy({
+      toPlayer: () => true,
+      byPlayer: () => true,
+    });
+  });
+  it('base energy regen', () => {
+    const val = energy.calcEnergyRegen([]);
+    expect(val).toBe(10);
+  });
+  it('energy regen with 0% haste', () => {
+    const val = energy.calcEnergyRegen([0]);
+    expect(val).toBe(10);
+  });
+  it('energy regen with -10% haste', () => {
+    const val = energy.calcEnergyRegen([-0.1]);
+    expect(val).toBe(9);
+  });
+  it('energy regen with bad parameter (int) value haste sets as base value', () => {
+    const val = energy.calcEnergyRegen(1);
+    expect(val).toBe(10);
+  });
+  it('energy regen with bad parameter (string) value haste sets as base value', () => {
+    const val = energy.calcEnergyRegen('it can\'t be this string value');
+    expect(val).toBe(10);
+  });
+  it('energy regen with bad parameter (object) value haste sets as base value', () => {
+    const val = energy.calcEnergyRegen(new Energy());
+    expect(val).toBe(10);
+  });
+  it('energy regen with 10% haste', () => {
+    const val = energy.calcEnergyRegen([0.1]);
+    expect(val).toBe(11);
+  });
+  it('energy regen with 1.52% haste', () => {
+    const val = energy.calcEnergyRegen([0.0152]).toFixed(2);
+    expect(val).toBe('10.15');
+  });
+  it('energy regen with 5.28% haste', () => {
```

Comment:
Running several tests with only values different; I think it would be cleaner to put this in an array and run a loop.